core/load-git.c

```diff
@@ -833,7 +833,7 @@ static void parse_settings_autogroup(char *line, struct membuffer *str, void *_u
 	(void) line;
 	(void) str;
 	(void) _unused;
-	set_autogroup(1);
+	set_autogroup(true);
```

Comment:
`core/parse-xml.c` also calls `set_autogroup()` based on a value from the XML `(zero / non-zero)`. this will probably just work silently, but should we also change `autogroupvalue;` to be `bool` in `try_to_match_autogroup()`, or keep the storage in `int`, but cast to `bool` in the `set_autogroup()` call?

Reply:
The code should work as is - the whole point of the `bool` type is to cast any integer to 0/1. A comment might be enough. On the other hand, this could be made more explicit by adding a `get_bool()` function. I added such a commit - your call.

Reply:
i think it makes sense to have this function, too. but maybe it should also handle the text `true/false` in the char buffer - otherwise it only handles `1/0`.

i would say - leave the `get_bool()` function out for now.

Reply:
...well let's keep it and handle the `true/false` eventually.